opdatasigverify.md

```diff
@@ -0,0 +1,122 @@
+# Enable oracle-based data import via OP_DATASIGVERIFY
+
+Version 0.2, 2018-02-22
+
+## Introduction
+
+OP_DATASIGVERIFY allows signed data to be imported into a script.  This data can then have many uses, depending on the rest of the script, such as deciding spendability of several possible addresses.  This opcode therefore enables the powerful blockchain concept of an "oracle" -- an entity that publishes authoritative statements about extra-blockchain events -- to be used in the Bitcoin Cash blockchain.  For an example use of how this opcode can be used to enable binary contracts on any security or betting on any quantitatively decidable event (such as a sports match) please [click here](https://medium.com/@g.andrew.stone/bitcoin-scripting-applications-decision-based-spending-8e7b93d7bdb9).  But this is just one example; as the Bitcoin Cash Script language grows in expressiveness, it is anticipated that this opcode will be used in many other applications.
+
+## Requirements on miners, full nodes, and clients
+
+Miners, full nodes, and clients will implement the OP_DATASIGVERIFY opcode and activate it during the May 2018 hard fork.
+
+## OP_DATASIGVERIFY Specification
+
+*OP_DATASIGVERIFY uses a new opcode number*
+
+    Opcode (decimal): 187
+    Opcode (hex): 0xbb
```

Comment:
Correct me if I'm wrong, but I thought the next unused OP_NOPx should be used instead? OP_NOP4 = 0xb3

Reply:
we can't use nop instructions because it is not a soft-forkable change... it manipulates the stack